src/core/syscalls/syscall_response.rs

```diff
@@ -123,6 +123,10 @@ impl WriteSyscallResponse for CallContractResponse {
             &(syscall_ptr + CallContractRequest::count_fields()),
             &self.retdata,
         )?;
+        vm.insert_value::<Felt>(
```

Comment:
Check the struct
struct CallContractResponse {
    retdata_size: felt,
    retdata: felt*,
}

retdata_size is inserted first